integration-tests/plugins/factories/simple-order-factory.ts

```diff
@@ -92,16 +100,44 @@ export const simpleOrderFactory = async (
     currency_code: currencyCode,
     tax_rate: taxRate,
     shipping_address_id: address.id,
-  })
+  }
+
+  let sc_id
+  if (isDefined(data.sales_channel)) {
+    let sc
+    
+    if (isString(data.sales_channel)) {
+      sc = await manager.findOne(SalesChannel, {
+        where: { id: data.sales_channel },
+      })
+    }
+
+    if (!sc) {
+      sc = await simpleSalesChannelFactory(
+        connection,
+        isString(data.sales_channel)
+          ? { id: data.sales_channel }
+          : data.sales_channel
+      )
+    }
+
+    sc_id = sc.id
+  }
+
+  if (sc_id) {
+    toCreate.sales_channel_id = sc_id
+  }
```

Comment:
todo: if `data.sales_channel`is defined but is not a string, we don't do anything. I am missing it?

Reply:
if `data.sales_channel` is defined but not a string we pass it into the sales channel factory on line 120

Reply:
LOL, I did not see the not in the if constraint. I need to clean my glasses